### PR TITLE
Adds handling to the path manager so it can deal with both looping and non-looping paths

### PR DESCRIPTION
**Pull request by Areloch**
Adds handling to the path manager so it can deal with both looping and non-looping paths
As provided by Ubiq from their Action Adventure Kit

### CODE REVIEW

File: Engine/source/scene/pathManager.cpp [hunk 1]
```diff
@@ -299,15 +303,33 @@ void PathManager::getPathPosition(const U32 id,
 
    // Ok, query holds our path information...
    F64 ms = msPosition;
-   if (ms > mPaths[id]->totalTime)
-      ms = mPaths[id]->totalTime;
+
+   //Looping vs. non-looping splines
+   F32 msTotal;
+   if (mPaths[id]->looping)
+      msTotal = mPaths[id]->totalTime;
```
Comment (lukaspj) [inline Engine/source/scene/pathManager.cpp:310]: I know that this is similar to how the code was written before, but I believe our official code-style is to always include braces?

File: Engine/source/scene/pathManager.cpp [hunk 2]
```diff
@@ -299,15 +303,33 @@ void PathManager::getPathPosition(const U32 id,
 
    // Ok, query holds our path information...
    F64 ms = msPosition;
-   if (ms > mPaths[id]->totalTime)
-      ms = mPaths[id]->totalTime;
+
+   //Looping vs. non-looping splines
```
Comment (lukaspj) [inline Engine/source/scene/pathManager.cpp:307]: Nit-picking here but I prefer having a space between // and comment text:

```suggestion
   // Looping vs. non-looping splines
```

File: Engine/source/scene/pathManager.cpp [hunk 3]
```diff
@@ -299,15 +303,33 @@ void PathManager::getPathPosition(const U32 id,
...
+   F32 msTotal;
```
Comment (lukaspj) [inline Engine/source/scene/pathManager.cpp:308]: Shouldn't this be a `U32` instead?

```suggestion
   U32 msTotal;
```

File: Engine/source/scene/pathManager.cpp [hunk 4]
```diff
@@ -430,5 +466,83 @@ bool PathManager::readState(BitStream* stream)
...
+   //2) Otherwise, divide the spline-segment into S sub-segments (S+1 points to test)
+   //2) Test these points against the given point and find the closest of them
+   //4) Recurse within the 2 segments surrounding the closest point
 
+   //NOTE: In a case where the spline comes near the point multiple times, the wrong local 
+   //minima of the spline may be chosen early on and then refined, like polishing a turd :)
+
+   PROFILE_START(PathManager_getClosestTimeToPoint);
+
+   F64 totalTime;
```
Comment (lukaspj) [inline Engine/source/scene/pathManager.cpp:485]:
```suggestion
   U64 totalTime;
```

**Comment by Azaezel** [comment]
went ahead and merged this because it looks like it contains a dependency for https://github.com/Torque3DResources/ActionAdventureKit and didn't display any particular nastiness for the testbed I'm using at least despite the concerns, though may indeed merit followup

**Pull request merged**